File: sdk/python/tempmail_sdk/providers/b_smelly_cc.py
"""
Mailinator 姊妹域名渠道：b.smelly.cc
MX 指向 mailinator，收信复用 mailinator 的 domain=public API。
"""

from typing import List

from ..types import Email, EmailInfo
from . import mailinator

CHANNEL = "b-smelly-cc"
DOMAIN = "b.smelly.cc"


def generate_email() -> EmailInfo:
    """生成 b.smelly.cc 域名的临时邮箱"""
    local = mailinator._random_string(12)
    return EmailInfo(channel=CHANNEL, email=f"{local}@{DOMAIN}")


def get_emails(token: str, email: str = "", **kwargs) -> List[Email]:
    """收信直接委托给 mailinator（domain=public 可读所有姊妹域名邮件）"""
    return mailinator.get_emails(token, email, **kwargs)

File: sdk/rust/src/providers/b_smelly_cc.rs
/*!
 * Mailinator 姊妹域名 b.smelly.cc
 * MX 指向 mail.mailinator.com，收信复用 mailinator provider 的读信逻辑。
 */

use crate::providers::mailinator;
use crate::types::{Channel, Email, EmailInfo};
use rand::Rng;

const DOMAIN: &str = "b.smelly.cc";

/// 生成随机本地部分（小写字母 + 数字）
fn random_local(len: usize) -> String {
    const CHARS: &[u8] = b"abcdefghijklmnopqrstuvwxyz0123456789";
    let mut rng = rand::thread_rng();
    (0..len)
        .map(|_| CHARS[rng.gen_range(0..CHARS.len())] as char)
        .collect()
}

/// 创建 b.smelly.cc 临时邮箱（Mailinator 姊妹域名，无需 token）
pub fn generate_email() -> Result<EmailInfo, String> {
    Ok(EmailInfo {
        channel: Channel::BSmellyCc,
        email: format!("{}@{}", random_local(12), DOMAIN),
        token: None,
        expires_at: None,
        created_at: None,
    })
}

/// 获取 b.smelly.cc 邮件列表（复用 mailinator 读信逻辑）
pub fn get_emails(email: &str) -> Result<Vec<Email>, String> {
    mailinator::get_emails(email)
}
